### DISCUSSION TITLE
Refreshing an entity

### DISCUSSION THREAD

**Discussion by edorgeville** [mikro-orm/mikro-orm#2720] [Q&A] [answered]
First of all, congratulations on the release of v5!
While reading the [release announcement](https://itnext.io/mikro-orm-5-stricter-safer-smarter-b8412e84cca4#c701), a question came to mind: is it possible to refresh an entity manually, without calling `findOne`? For example:
```typescript
const user = await User.findOne({name: 'John Doe'})
// Do stuff

// Database record was updated by another application, changing the name
// We force the refresh of the user
user.refresh() // ??
console.log(user.name) // New name from the DB
```
Thanks!

**Comment by B4nan** [comment, marked as answer]
What would you expect from that? Sync call obviously can't load any new data. And if reloading that entity is what you want, `findOne` is exactly the method to use. We can add `em.refresh()` (as well as `wrap(e).refresh()`, managed entities do have the EM reference), but it would be exactly the same thing, just a `findOne` call with `refresh: true`.

**Reply by edorgeville** [reply]
Gotcha, thanks 👍

**Reply by edorgeville** [reply]
@B4nan I have a similar issue now with a `@OneToMany` collection:
```typescript
const user = await User.findOne({name: 'John Doe'})
await user.books.loadItems()
console.log(user.books[0]) // { title: "Hello world" }

// Do stuff
// Another application updated the title to `Bye John`

const refreshedUser = await User.findOne({name: 'John Doe'}, {refresh: true})`
await refreshedUser.books.loadItems()
console.log(refreshedUser.books[0]) // { title: "Hello world" } <-- not updated
```
It seems that loaded relationships are not refreshed. I checked the [`loadItems()`](https://mikro-orm.io/api/next/core/class/Collection#loadItems) method but it doesn't seem to have a `refresh: true` option. 
Thanks in advance!

**Reply by B4nan** [reply]
Use `init()` instead, loadItems will never reload.

**Reply by edorgeville** [reply]
Thank you, I will try this out 🙌

**Comment by ddadaal** [comment]
This might be useful in tests. Many of my test cases has the following structure

```ts
it("tests a logic", async () => {
  // creates or loads an entity
  const user = await em.find(User, filter);
  await em.persistAndFlush(user);

  // call the server to change the state of the entity
  await service.changeState({ userId: user.id });

  // refresh the state of the entity
  // ???

  // validate the entity's current state
  expect(user.property).toBe(expected);
});
```

Currently I have to declare another variable and use findOne (as suggested in the answer) and a filter to search the db and fetch another instance of the entity.

In this situation it would be very convenient to have a refresh method to reload an existing entity.

**Reply by mattbdc** [reply]
just exploring this myself is `await wrap(entity).init()` the equiv of a reload?

**Comment by edorgeville** [comment]
Updating this: just discovered that this functionality was added in https://github.com/mikro-orm/mikro-orm/pull/3522 (see [corresponding docs](https://mikro-orm.io/docs/entity-manager#refreshing-entity-state)).
Thanks @B4nan !!
